pkg/asset/installconfig/azure/validation.go

```diff
@@ -102,9 +103,18 @@ func ValidateInstanceType(client API, fieldPath *field.Path, region, instanceTyp
 				errMsg := fmt.Sprintf("PremiumIO not supported for instance type %s", instanceType)
 				allErrs = append(allErrs, field.Invalid(fieldPath.Child("osDisk", "diskType"), diskType, errMsg))
 			}
+		} else if strings.EqualFold(*capability.Name, "UltraSSDAvailable") {
+			ultraSSDAvailable = strings.EqualFold(*capability.Value, "True")
 		}
 	}
 
+	// The UltraSSDAvailable capability might not be present at all, in which
+	// case it must assumed to be false
+	if ultraSSDEnabled && !ultraSSDAvailable {
+		errMsg := fmt.Sprintf("UltraSSD capability not supported for instance type %s", instanceType)
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
```

Comment:
Since we are already providing the instanceType value in the Invalid function, let's just mention it once?
```suggestion
		allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, "UltraSSD capability not supported for provided instance type"))
```